starknet-core/src/types/eth_address.rs

```diff
@@ -156,3 +168,75 @@ impl From<EthAddress> for FieldElement {
         FieldElement::from_byte_slice_be(&value.inner).unwrap()
     }
 }
+
+impl TryFrom<&[u8]> for EthAddress {
+    type Error = FromBytesSliceError;
+
+    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
+        if value.len() != 20 {
+            return Err(FromBytesSliceError(format!(
+                "expected 20 bytes found {}",
+                value.len()
+            )));
+        }
+
+        Ok(Self {
+            inner: value
+                .try_into()
+                .map_err(|e| FromBytesSliceError(format!("can't convert slice to array: {}", e)))?,
+        })
```

Comment:
No error mapping is needed here I think? I believe the only reason this can fail here is a length mismatch, which we already checked for.

Reply:
That's a very good point, will edit accordingly.

Reply:
Changed by unwrap with a comment on the unwrap safety.